platform/config/config-mapping-api/src/main/java/org/codice/ddf/config/mapping/ConfigMappingProvider.java

```diff
@@ -0,0 +1,110 @@
+   * provide all properties.
+   *
+   * <p>At least one provider capable of providing all properties and not just partial one is
+   * required to be bound to the mapping service before a corresponding mapping can be made
+   * available.
+   *
+   * @return <code>true</code> if this provider is only capable of providing parts of all the
+   *     properties for this mapping; <code>false</code> if it is able to provide all of them
+   */
+  public boolean isPartial();
+
+  /**
+   * Checks if this provider can provide mapped dictionaries for a given configuration mapping.
+   *
+   * <p><i>Note:</i> A provider is expected not to change which configuration mappings it provides
+   * for unless the provider is rebound with the {@link ConfigMappingService} which will re-compute
+   * which config mappings are impacted by this change.
+   *
+   * @param mapping the config mapping to check if this provider can provide for
+   * @return <code>true</code> if this provider can provide mapped dictionaries for the specified
+   *     config mapping; <code>false</code> otherwise
+   */
+  public default boolean canProvideFor(ConfigMapping mapping) {
+    return canProvideFor(mapping.getId());
+  }
+
+  /**
+   * Checks if this provider can provide mapped dictionaries for a given configuration mapping.
+   *
+   * <p><i>Note:</i> A provider is expected not to change which configuration mappings it provides
+   * for unless the provider is rebound with the {@link ConfigMappingService} which will re-compute
+   * which config mappings are impacted by this change.
+   *
+   * @param id the id of the config mapping to check if this provider can provide for
+   * @return <code>true</code> if this provider can provide mapped dictionaries for the specified
+   *     config mapping; <code>false</code> otherwise
+   */
+  public boolean canProvideFor(ConfigMapping.Id id);
+
+  /**
+   * Provides the mapped dictionary for a given configuration mapping.
+   *
+   * @param id the unique config mapping id for the mapped properties to provide
+   * @param config the configuration to use when computing the mapper properties
+   * @return a map corresponding to all mapped properties and their current values after
+   *     re-evaluating any internal rules defined
+   * @throws ConfigMappingException if a failure occurred while resolving this config mapping
+   */
+  public Map<String, Object> provide(ConfigMapping.Id id, ConfigService config)
+      throws ConfigMappingException;
+
+  /**
+   * {@inheritDoce}
+   *
+   * <p>The comparison must support the ranking priority described in the class description.
+   *
+   * @param provider the other provider to compare this provider with
+   * @return <code>1</code> if this provider has a higher ranking priority than the one provided;
+   *     <code>-1</code> if it has a lower ranking priority and <code>0</code> if they have equal
+   *     priorities
+   */
+  @Override
+  int compareTo(ConfigMappingProvider provider);
```

Comment:
✏️ Provide a default implementation that is based on the provided ranking by `getRanking()`

Reply:
Fixed